frontend/src/store/auth-context-provider.tsx

```diff
@@ -0,0 +1,35 @@
+import {ReactNode, useReducer} from "react";
+import {AuthContext, AuthContextProps} from "./auth-context";
+
+enum ActionType {
+  LOGIN = 'login',
+  LOGOUT = 'logout'
+}
```

Comment:
You should (almost) always prefer objects marked with `as const` to Typescript's enum. In this case it does not really matter because action type is not used outside of this file and everything works as expected. However, if we were to use an enum in multiple cases in the project, we could stumble upon weird behaviours and typing issues.

Have a read, if you have not known about it before:

https://www.crocoder.dev/blog/typescript-enums-good-bad-and-ugly/

https://graphqleditor.com/blog/enums-are-still-bad/

tldr: typescript enum bad, use as const

```typescript
const ActionTypes = {
  LOGIN: "login",
  LOGOUT: "logout"
} as const;

type ActionType = typeof ActionTypes[keyof typeof ActionTypes];
```

Reply:
I've heard that TS enums are bad, but never encountered any problems.

keyof typeof looks kinda funny, but i am open to learn some of TS magic 👍